src/KM_MapEditor.pas

```diff
@@ -355,10 +396,6 @@ procedure TKMMapEditor.MouseUp(Button: TMouseButton; aOverMap: Boolean);
                                 //Actual change was made in UpdateStateIdle, we just register it is done here
                                 fTerrainPainter.MakeCheckpoint;
                               end;
-                cmObjects:    begin
-                                gTerrain.Land[P.Y,P.X].Obj := 255; //Delete object
-                                fTerrainPainter.MakeCheckpoint;
-                              end;

```

Comment:
Ah sorry, didn't realise that was only for right clicks. Yeah right clicking in object mode doesn't need to do anything here.